packages/extension-koni-ui/src/Popup/Home/Games/gameSDK.ts

```diff
@@ -282,14 +291,40 @@ export class GameApp {
   }
 
   onShowLeaderboard () {
-    console.log('show leaderboard');
+    window.location.href = '/home/leaderboard';
   }
 
   onShowShop () {
     console.log('open shop');
   }
 
-  onGetLeaderboard (req: GetLeaderboardRequest): GetLeaderboardResponse {
+  async onGetLeaderboard (req: GetLeaderboardRequest): Promise<GetLeaderboardResponse> {
+    const leaderboardGeneral = this.apiSDK.leaderboardConfig.leaderboard_general as unknown as LeaderboardGroups[];
+    const leaderboards = this.apiSDK.leaderboardConfig.leaderboard_map as unknown as LeaderboardInfo[];
+
+    if (leaderboardGeneral && leaderboards) {
+      const firstLeaderboardGroups = leaderboardGeneral[0];
+
+      if (!firstLeaderboardGroups || !firstLeaderboardGroups.leaderboards.length) {
+        return { players: [] };
+      }
+
+      const weekLeaderBoard = leaderboards.find((l) => l.id === firstLeaderboardGroups.leaderboards[0]?.id);
```

Comment:
This may not get the right leaderboard, please find leaderboard with type `mythical:event:point` and specialTime is weekly